libexec/pyenv-exec

```diff
@@ -41,7 +43,7 @@ for script in "${scripts[@]}"; do
 done
 
 shift 1
-# CPython's `sys.executable` requires the `PYENV_BIN_PATH` to be at the top of the `PATH`.
-# https://github.com/pyenv/pyenv/issues/98
-export PATH="${PYENV_BIN_PATH}:${PATH}"
-exec -a "$PYENV_COMMAND" "$PYENV_COMMAND_PATH" "$@"
+if [ "$pyenv_command_version" != "system" ]; then
```

Comment:
As per my understanding, you needed to add the code around "PYENV_DISPLAY_VERSION" to didtinguish if the executing command is derived from system or not. How about checking if "PYENV_COMMAND_PATH" is starting from "PYENV_ROOT" or not...? With it, I think you don't need to modifu pyenv-which's behavior. How do you think?

Reply:
Awesome, pushed a fixup.